qutip/random_objects.py

```diff
@@ -319,13 +319,22 @@ def rand_ket(N=0, density=1, dims=None, seed=None):
     oper : qobj
         Nx1 ket state quantum operator.
 
+    Raises
+    -------
+    ValueError
+        Specify either the number of rows of operator (N) or
+        dimensions of quantum object (dims).
+
     """
     if seed is not None:
         np.random.seed(seed=seed)
-    if N and dims:
+    if N is None and dims is None:
+        raise ValueError('Specify either the number of rows of operator'
+                         '(N) or dimensions of quantum object (dims)')
```

Comment:
I was going to question why this refers to an operator, but I just saw the docstring of this function has it wrong as well.  Could you change this exception (and the one in `rand_ket_haar`), and the docstring of this function to refer to a "state vector" instead of an "operator"?